src/util/searchSubmissions.js

```diff
@@ -1,6 +1,38 @@
 import firebase from 'firebase';
 import StudentSubmission from './StudentSubmission';
 
+// expands a date range by 1 month on either side
+// this deals with messy time offsets when pulling from Firestore
+// TODO: find a better way to deal with time offsets!
+function cleanDates(start, end) {
+    const startMonth = start.getMonth();
+    const endMonth = end.getMonth();
+
+    let expStart, expEnd;
+
+    if (startMonth > 0) {
+        expStart = new Date(
+            start.getFullYear(), startMonth - 1, start.getDate()
+        );
+    } else {
+        expStart = new Date(
+            start.getFullYear() - 1, 11, start.getDate()
```

Comment:
u can definitely refactor this code to be way easier

`startMonth > 0 ? (someNewVariable = startMonth - 1, year = start.getFullYear()) : (someNewVariable = 11, year = start.getFullYear() - 1)
let expStart = new Date(year, someNewVariable, start.getDate())'

I don't really get what this is doing though - we have year, month, date and why would there be an input == 0 ever? Is that even valid - I think we should not be able to let users put in an invalid month outside of the range 1-12 but maybe I'm misunderstanding